src/Test.UnitTests.BinSkim.Driver/CompilerDataLoggerTests.cs

```diff
@@ -622,6 +631,139 @@ private void LogExceptionEvents(List<ExecutionException> exceptions, CompilerDat
             return sb;
         }
 
+        private StringBuilder ValidateChunkedEvents(string expectedEventName,
+                                                    List<ITelemetry> telemetryGeneratedEvents,
+                                                    string expectedChunkedContent,
+                                                    int chunkCount)
+        {
+            var sb = new StringBuilder();
+            string assemblyEventsId = string.Empty;
+            string commandLineEventsId = string.Empty;
+
+            var compilerEvents = new List<EventTelemetry>();
+            var assemblyReferencesEvents = new List<EventTelemetry>();
+            var commandLineEvents = new List<EventTelemetry>();
+            var summaryEvents = new List<EventTelemetry>();
+
+            foreach (EventTelemetry telemetryEvent in telemetryGeneratedEvents)
+            {
+                IDictionary<string, string> properties = telemetryEvent.Properties;
+
+
+                if (string.IsNullOrWhiteSpace(commandLineEventsId)
+                    && properties.TryGetValue(CompilerDataLogger.CommandLineId, out string commandLineEventId))
+                {
+                    commandLineEventsId = commandLineEventId;
+                }
+
+                if (string.IsNullOrWhiteSpace(assemblyEventsId)
+                    && properties.TryGetValue(CompilerDataLogger.AssemblyReferencesId, out string assemblyEventId))
+                {
+                    assemblyEventsId = assemblyEventId;
+                }
+
+                switch (telemetryEvent.Name)
+                {
+                    case CompilerDataLogger.CompilerEventName:
+                    {
+                        compilerEvents.Add(telemetryEvent);
+
+                        if (expectedEventName == CompilerDataLogger.AssemblyReferencesEventName
+                            && !properties.ContainsKey(CompilerDataLogger.AssemblyReferencesId))
+                        {
+                            sb.AppendLine("Compiler Event is missing the `assemblyReferencesId`");
+                        }
+                        else if (expectedEventName == CompilerDataLogger.CommandLineEventName
+                            && !properties.ContainsKey(CompilerDataLogger.CommandLineId))
+                        {
+                            sb.AppendLine("Compiler Event is missing the `commandLineId`");
+                        }
+                        break;
+                    }
+                    case CompilerDataLogger.AssemblyReferencesEventName:
+                    {
+                        assemblyReferencesEvents.Add(telemetryEvent);
+
+                        if (expectedEventName == CompilerDataLogger.AssemblyReferencesEventName
+                            && !expectedChunkedContent.Contains(properties[CompilerDataLogger.ChunkedAssemblyReferences]))
```

Comment:
>&& !expectedChunkedContent.Contains(properties[CompilerDataLogger.ChunkedAssemblyReferences])) [](http://example.com/codeflow?start=28&length=94)

Have not check this file before, I ran the test case locally and see the behavior,

expectedChunkedContent:
"Microsoft.DiaSymReader (1.3.0);Newtonsoft.Json (13.0.1)"

properties[CompilerDataLogger.ChunkedAssemblyReferences]:
"DiaSymRead"

This condition will pass the contains will be true, 
wondering why it is "DiaSymRead" not "DiaSymReader" or "Microsoft.DiaSymReader (1.3.0)"
do you know,

Reply:
ok! I see now, DiaSymRead is split from the whole string and we are comparing using contains.